greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchCommandParser.java

```diff
@@ -0,0 +1,87 @@
+package com.icegreen.greenmail.imap.commands;
+
+import com.icegreen.greenmail.imap.ImapRequestLineReader;
+import com.icegreen.greenmail.imap.ProtocolException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.mail.search.AndTerm;
+import javax.mail.search.NotTerm;
+import javax.mail.search.SearchTerm;
+
+/**
+ * Created on 10/03/2016.
+ *
+ * @author Reda.Housni-Alaoui
+ */
```

Comment:
Hate to sound pedantic but all you did here was split out SearchCommandParser which was an inner class in SearchCommand. That doesn't exactly warrant putting your name here as the author of this class. 

The actual author was shown in the following comment block at the top of SearchCommand

```
/**
  * Handles processing for the SEARCH imap command.
  *
  * @author Darrell DeBoer <darrell@apache.org>
  */
```

We need to maintain this comment as well: 
```
 Original file can be found at http://james.apache.org
```